cmd/ctl/pkg/renew/renew.go

```diff
@@ -0,0 +1,249 @@
+
+	return cmd
+}
+
+// Validate validates the provided options
+func (o *Options) Validate(cmd *cobra.Command, args []string) error {
+	if len(o.LabelSelector) > 0 && len(args) > 0 {
+		return errors.New("cannot specify Certificate arguments as well as label selectors")
+	}
+
+	if o.All && len(args) > 0 {
+		return errors.New("cannot specify Certificate arguments as well as --all flag")
+	}
+
+	return nil
+}
+
+// Complete takes the command arguments and factory and infers any remaining options.
+func (o *Options) Complete(f cmdutil.Factory, cmd *cobra.Command, args []string) error {
+	var err error
+	o.Namespace, _, err = f.ToRawKubeConfigLoader().Namespace()
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// Run executes renew command
+func (o *Options) Run(f cmdutil.Factory, cmd *cobra.Command, args []string) error {
+	restConfig, err := f.ToRESTConfig()
+	if err != nil {
+		return err
+	}
+
+	cmClient, err := cmclient.NewForConfig(restConfig)
+	if err != nil {
+		return err
+	}
+
+	nss := []corev1.Namespace{{ObjectMeta: metav1.ObjectMeta{Name: o.Namespace}}}
+
+	// TODO: handle network context
+
+	if o.AllNamespaces {
+		kubeClient, err := kubernetes.NewForConfig(restConfig)
+		if err != nil {
+			return err
+		}
+
+		nsList, err := kubeClient.CoreV1().Namespaces().List(context.TODO(), metav1.ListOptions{})
+		if err != nil {
+			return err
+		}
+
+		nss = nsList.Items
+	}
+
+	var crts []cmapi.Certificate
+	for _, ns := range nss {
+		switch {
+		case o.All, len(o.LabelSelector) > 0:
+			crtsList, err := cmClient.CertmanagerV1alpha2().Certificates(ns.Name).List(context.TODO(), metav1.ListOptions{
+				LabelSelector: o.LabelSelector,
+			})
+			if err != nil {
+				return err
+			}
+
+			crts = append(crts, crtsList.Items...)
+
+		default:
+			for _, crtName := range args {
+				crt, err := cmClient.CertmanagerV1alpha2().Certificates(ns.Name).Get(context.TODO(), crtName, metav1.GetOptions{})
+				if err != nil {
+					return err
+				}
+
+				crts = append(crts, *crt)
+			}
+		}
+	}
+
+	if len(crts) == 0 {
+		if o.AllNamespaces {
+			fmt.Fprintln(o.ErrOut, "No Certificates found")
+		} else {
+			fmt.Fprintf(o.ErrOut, "No Certificates found in %s namespace.\n", o.Namespace)
+		}
+
+		return nil
+	}
+
+	for _, crt := range crts {
+		if err := o.renewCertificate(cmClient, &crt); err != nil {
+			return err
+		}
+	}
+
+	if o.Wait {
+		if err := o.waitCertificatesReady(cmClient, crts); err != nil {
+			return err
+		}
+
+		fmt.Fprintf(o.Out, "%d Certificates successfully renewed\n", len(crts))
+	}
+
+	return nil
+}
+
+func (o *Options) renewCertificate(cmClient *cmclient.Clientset, crt *cmapi.Certificate) error {
+	apiutil.SetCertificateCondition(crt, cmapi.CertificateConditionIssuing, cmmeta.ConditionTrue, "ManuallyTriggered", "Certificate re-issuance manually triggered")
+	_, err := cmClient.CertmanagerV1alpha2().Certificates(crt.Namespace).UpdateStatus(context.TODO(), crt, metav1.UpdateOptions{})
```

Comment:
Can you include a `ctx` arg to this method and use it here? Ideally there should only be a single call to `context.TODO()` at the start of the command, and that context should be passed through to all methods and used here.

Reply:
:+1: Will pipe through